tests/python/gaia-ui-tests/gaiatest/gaia_graphics_test.py

```diff
@@ -109,30 +120,40 @@ def take_screenshot(self):
             else:
                 error_msg = "Ref file not found for: " + filename + '\n'
             if error_msg != "":
-                self.logger.critical(error_msg)
+                #self.logger.critical(error_msg)  uncomment for debugging
                 self.failcomment += error_msg
                 self.test_passed = False
 
         self.picture_index += 1
 
-    def image_compare(self, target_img, ref_img, diff_img, fuzz_value):
+    def image_compare(self, target, ref, diff, fuzz_value):
         """do single image compare using the convert console command of ImageMagick"""
 
         message = ""
 
         p = subprocess.Popen(
-            ["compare", "-fuzz", str(fuzz_value) + "%", "-metric", "AE", target_img, ref_img, diff_img],
+            ["compare", "-fuzz", str(fuzz_value) + "%", "-metric", "AE", target, ref, diff],
             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         out, err = p.communicate()
         p.wait()
 
         if not (err == '0\n' or err == '0'):
             err = err.replace('\n', '')
-            message = 'WARNING: ' + err + ' pixels mismatched between ' + target_img + ' and ' + ref_img + '\n'
+            message = 'WARNING: ' + err + ' pixels mismatched between ' + target + ' and ' + ref + '\n'
+
+            # move the diff image and the target image to a separate folder
+            if not os.path.exists(self.mismatch_path):
+                os.makedirs(self.mismatch_path)
+            target_image_name = target[target.rfind("/")+1:]
+            diff_image_name = diff[diff.rfind("/") + 1:]
+            os.rename(target, os.path.join(self.mismatch_path, target_image_name))
+            os.rename(diff, os.path.join(self.mismatch_path, diff_image_name))
 
         return message
 
-    def edge_scroll(self, frame, direction, dist, release=True):
+    # Make UI action related methods static, so they can be used outside the GaiaImageCompareTestCase object as well.
```

Comment:
I don't understand. Why are these methods in gaia_graphics_test.py then? Why not in gaia_test.py?